api/internal/handler/route_export/route_export.go

```diff
@@ -0,0 +1,349 @@
+
+const (
+	BasicAuth AuthType = "basic-auth"
+	KeyAuth   AuthType = "key-auth"
+	JWTAuth   AuthType = "jwt-auth"
+)
+
+var (
+	openApi = "3.0.0"
+	title   = "Routes Export"
+)
+
+func routeToOpenApi3(routes []*entity.Route) *openapi3.Swagger {
+	paths := openapi3.Paths{}
+	pathItem := &openapi3.PathItem{}
+	path := &openapi3.Operation{}
+	paramsRefs := []*openapi3.ParameterRef{}
+	requestBody := &openapi3.RequestBody{}
+	components := &openapi3.Components{}
+	secSchemas := openapi3.SecuritySchemes{}
+	for _, route := range routes {
+
+		extensions := make(map[string]interface{})
+		path.Summary = route.Desc
+		path.OperationID = route.Name
+		if route.URI != "" {
+			extensions["x-apisix-uri"] = route.URI
+		}
+		if route.Upstream != nil {
+			extensions["x-apisix-upstream"] = route.Upstream
+		}
+		if route.Host != "" {
+			extensions["x-apisix-host"] = route.Host
+		}
+		if route.Hosts != nil {
+			extensions["x-apisix-hosts"] = route.Hosts
+		}
+		if route.Labels != nil {
+			extensions["x-apisix-labels"] = route.Labels
+		}
+		if route.RemoteAddr != "" {
+			extensions["x-apisix-remoteAddr"] = route.RemoteAddr
+		}
+		if route.RemoteAddrs != nil {
+			extensions["x-apisix-remoteAddrs"] = route.RemoteAddrs
+		}
+		if route.FilterFunc != "" {
+			extensions["x-apisix-filterFunc"] = route.FilterFunc
+		}
+		if route.ServiceID != nil {
+			extensions["x-apisix-serviceID"] = route.ServiceID
+		}
+		if route.UpstreamID != nil {
+			extensions["x-apisix-upstreamID"] = route.UpstreamID
+		}
+		if route.Script != nil {
+			extensions["x-apisix-script"] = route.Script
+		}
+		if route.ServiceProtocol != "" {
+			extensions["x-apisix-serviceProtocol"] = route.ServiceProtocol
+		}
+		if route.Vars != nil {
+			extensions["x-apisix-vars"] = route.Vars
+		}
+		// analysis route.URIs
+		for _, uri := range route.Uris {
+			if strings.Contains(uri, "*") {
+				paths[strings.Split(uri, "*")[0]+"{params}"] = pathItem
+				// add params introduce
+				paramsRefs = append(paramsRefs, &openapi3.ParameterRef{
+					Value: &openapi3.Parameter{
+						In:          "path",
+						Name:        "params",
+						Required:    true,
+						Description: "params in path",
+						Schema:      &openapi3.SchemaRef{Value: &openapi3.Schema{Type: "string"}}}})
+			} else {
+				paths[uri] = pathItem
+			}
+		}
+		if route.Plugins != nil {
+			param := &openapi3.Parameter{}
+			secReq := &openapi3.SecurityRequirements{}
+			plugins := make(map[string]interface{})
+			// analysis plugins
+			for key, value := range route.Plugins {
+				// analysis proxy-rewrite plugin
+				//if key == "proxy-rewrite" {
+				//	continue
+				//}
+
+				// analysis request-validation plugin
+				if key == "request-validation" {
+					for k, v := range value.(map[string]interface{}) {
+						if k == "header_schema" {
+							param.In = "header"
+							requestValidation := &entity.RequestValidation{}
+							reqBytes, _ := json.Marshal(&v)
+							err := json.Unmarshal(reqBytes, requestValidation)
```

Comment:
@Jaycean Additional check is also required here.

```suggestion
							err := json.Unmarshal(reqBytes, requestValidation)
							if err != nil {
							        // log error and return back to client
							}
```

Reply:
Yes, I just added it

Reply:
Thks for checking.